### PR TITLE
Add success msg for only notes output

### LINKED ISSUE #12302: Success message missing when reveal_type() used.
**Bug Report**

I have something like `if TYPE_CHECKING: reveal_type(foo)` in a module (in 5 places).
mypy finds no errors, but the line that says "success" is missing.

Here's two sets of output from mypy, first with reveal_type() in the module, and second without.

```
D:\Solutions\trunk\Python\Sass>mypy .
solver\common.py:299: note: Revealed type is "Any"
solver\common.py:366: note: Revealed type is "Any"
solver\common.py:521: note: Revealed type is "solver.instruct.InstrTab"
solver\common.py:522: note: Revealed type is "Any"
solver\common.py:523: note: Revealed type is "Any"

D:\Solutions\trunk\Python\Sass>mypy .
Success: no issues found in 52 source files
```

What I think is correct output, in the first case, is
```
D:\Solutions\trunk\Python\Sass>mypy .
solver\common.py:299: note: Revealed type is "Any"
solver\common.py:366: note: Revealed type is "Any"
solver\common.py:521: note: Revealed type is "solver.instruct.InstrTab"
solver\common.py:522: note: Revealed type is "Any"
solver\common.py:523: note: Revealed type is "Any"
Success: no issues found in 52 source files
```

**Your Environment**

- Mypy version used: 0,92`
- Python version used: 3.7
- Operating system and version: Windows 10

### PR DESCRIPTION
**Pull request by 97littleleaf11**
Closes #12302 

The `messages` here could be neither error nor note, thus I modify the `count_stats ` to also count `note`.

### CODE REVIEW

**Comment by jhance** [comment]
It seems like this could lead to accidentally leaving a `reveal_type` in the codebase, are we sure failing isn't the right behavior?

File: mypy/main.py [hunk 1]
```diff
@@ -114,13 +114,16 @@ def main(script_path: Optional[str],
         code = 2 if blockers else 1
     if options.error_summary:
         if messages:
-            n_errors, n_files = util.count_stats(messages)
+            n_errors, n_notes, n_files = util.count_stats(messages)
```
Comment (jhance) [inline mypy/main.py:117]: I don't think the outer `if` remains necessary. If `messages` is empty, then this returns `(0, 0, 0)`. So we can just always count the stats and then condition on them.

Also, is `n_notes == len(messages)` actually different from `not n_errors`? Is there a third type of message that is neither an error nor a note?

Comment (97littleleaf11) [reply mypy/main.py:117]: There is a test case [testErrorSummaryOnBadUsage], which generates 'mypy: cannot read file 'missing.py': No such file or directory'

Comment (jhance) [reply mypy/main.py:117]: Okay, I still think the comment about not needing the outer `if` applies.

**Pull request merged**